app/src/main/java/gateway/services/ServiceMetadata.java

```diff
@@ -17,16 +17,17 @@ public static class ResSaveFile extends ResStatus
 		public UUID fileUUID;
 	}
 
-	public static ResSaveFile
-	saveFile (UUID userUUID, UUID directoryUUID, String filetype, String filename, int filesize)
+	public static ResSaveFile saveFile (
+		UUID userUUID, UUID directoryUUID, Boolean isFile, String filetype, String filename,
+		int filesize)
 	{
 
 		ServiceMetadata.ResSaveFile s = new ServiceMetadata.ResSaveFile ();
 		JSONObject body = new JSONObject ();
 
 		body.put ("userUUID", userUUID);
 		body.put ("parentUUID", directoryUUID == null ? JSONObject.NULL : directoryUUID);
-		body.put ("fileType", "archive");
+		body.put ("fileType", isFile ? "archive" : "directory");
 		body.put ("fileName", filename);
 		body.put ("fileExtension", filetype == null ? JSONObject.NULL : filetype);
```

Comment:
I suggest you to change fileType for fileExtension in this case, what do you think?